File: scripts/plot/lind_ham/les.py
import pandas as pd
from scripts.plot.routines.save import save_figure
from scripts.plot.routines.scatter import add_scatter_trace
from scripts.plot.routines.layout import add_layout
from scripts.plot.routines.violin import add_violin_trace
import os
import plotly.graph_objects as go

path = f"E:/YandexDisk/Work/os_lnd/draft/mbl/2/figures/lndham/lambda/100_100"

N = 100
alpha = 0.5
n_seeds = 50
T = 1.0

lpn_type = -1 # energy
lpn_log_deltas = [-6.0, -5.0, -4.0]
Ts = [1, 2, 5]

vio = go.Figure()
for T in Ts:
    for log_delta in lpn_log_deltas:
        fn = f"{path}/lambda_N({N})_numSeeds({n_seeds})_alpha({alpha:0.4f})_T({T:0.4f})_lpn({lpn_type}_{log_delta:0.4f}).csv"
        les = pd.read_csv(fn, header=None)
        les['mean'] = les.mean(axis=1)
        les['period'] = fr"Period={T}"
        if T == 1:
            showlegend = True
        else:
            showlegend = False
        add_violin_trace(vio, les.loc[:, 'mean'], fr"$\log_{{{10}}}\Delta={{{log_delta}}}$", x=les.loc[:, 'period'], showlegend=showlegend)

add_layout(vio, "", fr"$\lambda$", f"")
vio.update_layout({'colorway': ['blue', 'red', 'green']})
save_figure(vio, f"{path}/lambda_N({N})_numSeeds({n_seeds})_alpha({alpha:0.4f})_lpn({lpn_type})_vio")

File: scripts/plot/lind_ham/les.m
clear all;
addpath('../routines/matlab')

N = 100;
alpha = 0.5;
n_seeds = 50;

lpn_type = -1;
lpn_log_deltas = [-6.0, -5.0, -4.0]';
Ts = [1, 2, 5]';

path = 'E:/YandexDisk/Work/os_lnd/draft/mbl/2/figures/lndham/lambda/100_100';

fig = figure;

for log_delta_id = 1:size(lpn_log_deltas, 1)
    fig = figure;
    for T_id = 1:size(Ts, 1)
        fn = sprintf('%s/lambda_N(%d)_numSeeds(%d)_alpha(%0.4f)_T(%0.4f)_lpn(%d_%0.4f).csv', ...
            path, ...
            N, ...
            n_seeds, ...
            alpha, ...
            Ts(T_id), ...
            lpn_type, ...
            lpn_log_deltas(log_delta_id));
        
        data = importdata(fn);
        lambdas = data(:);
        
        pdf.x_num_bins = 100;
        pdf.x_label = '$\lambda$';
        pdf.x_bin_s = min(lambdas);
        pdf.x_bin_f = max(lambdas);
        pdf = pdf_1d_setup(pdf);
        pdf = pdf_1d_update(pdf, lambdas);
        pdf = pdf_1d_release(pdf);
        h = plot(pdf.x_bin_centers, pdf.pdf, 'LineWidth', 2);
        legend(h, sprintf('$\\tau=%d$', Ts(T_id)))
        legend(gca,'off');
        legend('Interpreter', 'latex');
        set(gca, 'FontSize', 30);
        xlabel(pdf.x_label, 'Interpreter', 'latex');
        set(gca, 'FontSize', 30);
        ylabel('$PDF$', 'Interpreter', 'latex');
        hold all;
    end
    fn_fig = sprintf('%s/density_lambda_N(%d)_numSeeds(%d)_alpha(%0.4f)_lpn(%d_%0.4f)', ...
        path, ...
        N, ...
        n_seeds, ...
        alpha, ...
        lpn_type, ...
        lpn_log_deltas(log_delta_id));
    save_fig(fig, fn_fig);
end
